app/services/credits/applied_coupon_service.rb

```diff
@@ -40,9 +41,13 @@ def already_applied?
     end
 
     def compute_amount
-      return invoice.amount_cents if remaining_amount > invoice.amount_cents
+      if applied_coupon.coupon.percentage?
+        discounted_value = invoice.amount_cents * applied_coupon.percentage_rate.fdiv(100)
 
-      remaining_amount
+        return (discounted_value >= invoice.amount_cents) ? invoice.amount_cents : discounted_value.round
+      end
+
+       applied_coupon.recurring? ? calculate_recurring_amount : calculate_one_time_coupon_amount
```

Comment:
Layout/IndentationConsistency: Inconsistent indentation detected. (https://rubystyle.guide#spaces-indentation, https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#follow-the-coding-conventions)